### DISCUSSION TITLE
Stale filing data when running in Fast API?

### DISCUSSION THREAD

**Discussion by martinnormark** [dgunning/edgartools#125] [General]
I have a Fast API application that I host on Railway. It has a few simple endpoints wrapping the functionality from this repo (awesome library!).

I poll the [recent filings feed](https://www.sec.gov/cgi-bin/browse-edgar?action=getcurrent&type=8-k&company=&dateb=&owner=include&start=0&count=40&output=atom) from SEC at an interval, and then call my API endpoint to get filing details by accession number. 

The API wrapper looks like this:

```python
from fastapi import APIRouter, Query, HTTPException
from edgar import Filing as EdgarFiling, get_filings, get_by_accession_number

router = APIRouter()

@router.get("/{accessionNumber}", response_model=Filing)
async def read_filing(accessionNumber: str):
    filing = get_by_accession_number(accessionNumber)

    if not filing:
        raise HTTPException(status_code=404, detail="Filing not found")

    return Filing(
        form=filing.form,
        accession_number=filing.accession_number,
        cik=filing.cik,
        company_name=filing.company,
        filing_date=filing.filing_date,
        url=filing.url,
        document_url=filing.document.url,
        text_url=filing.text_url,
    )
```

After my API have been running for a day or so, it begins to throw 404 from the above code. So an accession number e.g. `0001493152-24-041216` is not found.

I then test the same via a Colab notebook and it returns just fine. Then I restart the service on Railway and when it is back up, it can get the filing just fine.

Could there be any caching or other mechanism that could cause this?

**Comment by dgunning** [comment]
The function is defined as

```python
@lru_cache(maxsize=16)
def get_by_accession_number(accession_number: str, show_progress: bool = True):
```

I'm not sure exactly what could be causing the issue but I'm refactoring and testing that the function won't be cached whenever it returns None

**Reply by martinnormark** [reply]
Ok, it might be related to the host instead. I tried setting auto-sleep on and it seemed to help, as it would clear its state every once in a while.
